internal/kubernetes/virtualNodeupdater.go

```diff
@@ -42,9 +44,31 @@ func (p *KubernetesProvider) StartNodeUpdater(nodeRunner *node.NodeController) (
 		for {
 			select {
 			case ev := <-advWatcher.ResultChan():
-				p.ReconcileNodeFromAdv(ev)
+				err = p.ReconcileNodeFromAdv(ev)
+				if err != nil {
+					klog.Error(err)
+					advWatcher.Stop()
```

Comment:
Did you verify that this `Stop()` statement doesn't crash? Because I think that it could cause panic caused by a "closing of already closed channel" error. In that case I would add a `recover()` statement ([example](https://play.golang.org/p/LqfFbAV0cg1))

Reply:
Good point, thank you! I think it would be a very strange behaviour, since we are in the switch case where the channel received an event (so it was open) and before the `Stop()` statement nobody closes it. 
However I think that if that happens we don't have a way to recover the error and could only log a fatal error; do you have a possible recovery?

Reply:
Since we are re-opening the channel due to its close from the API server, I think that it is very likely to happen. maybe we could implement a function that re-open the chan and call it whenever the recover routine is triggered (just an idea based on the example that I linked in the comment above).

Reply:
I have written a recover function but while debugging I found out the `Stop()` function already checks if the channel is closed ([doc](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/watch/streamwatcher.go#L81)), so I think we don't need the recover; however, if it can be useful I commit it ( it's very simple, it siply restart the nodeUpdater)

Reply:
If `Stop()` already handles this error, I'm perfectly fine with your solution!